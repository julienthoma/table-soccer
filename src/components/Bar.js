import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { BLUE, GREY } from './Colors';

const Container = styled.div`
  position: relative;
  display: flex;
  width: 100%;
`;

const Part = styled.div`
  position: relative;
  background: ${props => props.color};
  height: 20px;
`;

const LeftPart = styled(Part)`
  width: ${props => props.width};
`;

const RightPart = styled(Part)`
  flex-grow: 1;
`;

const Text = styled.div`
  position: absolute;
  top: 3px;
  text-transform: uppercase;
  font-size: 13px;
`;

const LeftText = styled(Text)`
  left: 10px;
`;

const RightText = styled(Text)`
  right: 10px;
`;

const Bar = ({
  leftText,
  rightText,
  leftColor,
  rightColor,
  widthInPercent
}) => (
  <Container>
    <LeftPart width={widthInPercent} color={leftColor} />
    <LeftText>{leftText}</LeftText>
    <RightPart color={rightColor} />
    <RightText>{rightText}</RightText>
  </Container>
);

Bar.defaultProps = {
  leftText: '',
  rightText: '',
  leftColor: BLUE,
  rightColor: GREY
};

Bar.propTypes = {
  leftText: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  rightText: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  leftColor: PropTypes.string,
  rightColor: PropTypes.string,
  widthInPercent: PropTypes.string.isRequired
};

export default Bar;
