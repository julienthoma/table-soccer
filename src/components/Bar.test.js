import React from 'react';
import { renderWithMuiTheme } from '../test.utils';
import Bar from './Bar';

it('Bar renders correctly', () => {
  const bar = (
    <Bar
      leftText="Hello"
      rightText="World"
      widthInPercent="50%"
      leftColor="red"
      rightColor="green"
    />
  );
  expect(renderWithMuiTheme(bar)).toMatchSnapshot();
});
