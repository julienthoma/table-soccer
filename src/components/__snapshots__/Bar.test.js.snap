// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar renders correctly 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c1 {
  position: relative;
  background: red;
  height: 20px;
  width: 50%;
}

.c3 {
  position: relative;
  background: green;
  height: 20px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c2 {
  position: absolute;
  top: 3px;
  text-transform: uppercase;
  font-size: 13px;
  left: 10px;
}

.c4 {
  position: absolute;
  top: 3px;
  text-transform: uppercase;
  font-size: 13px;
  right: 10px;
}

<div
  className="c0"
>
  <div
    className="c1"
    color="red"
    width="50%"
  />
  <div
    className="c2"
  >
    Hello
  </div>
  <div
    className="c3"
    color="green"
  />
  <div
    className="c4"
  >
    World
  </div>
</div>
`;
